src/test/java/com/mongodb/InterruptibilityItTest.java

```diff
@@ -0,0 +1,314 @@
+        assertThrowsInVirtualInterruptedThread(MongoInterruptedException.class,
+                interruptDelay, "drop", collection::drop);
+    }
+
+    @ParameterizedTest
+    @MethodSource("arguments")
+    void mongoIterableFirst(final Duration interruptDelay) {
+        MongoIterable<String> iterable = client.listDatabaseNames();
+        assertThrowsInVirtualInterruptedThread(MongoInterruptedException.class,
+                interruptDelay, "listDatabases", iterable::first);
+    }
+
+    @ParameterizedTest
+    @MethodSource("arguments")
+    void mongoCursorNext(final Duration interruptDelay) {
+        collection.insertMany(IntStream.range(0, 2).mapToObj(i -> new Document()).toList());
+        try (MongoCursor<Document> cursor = collection.find().batchSize(1).cursor()) {
+            com.mongodb.assertions.Assertions.assertTrue(cursor.available() == 1);
+            cursor.next();
+            com.mongodb.assertions.Assertions.assertTrue(cursor.available() == 0);
+            assertThrowsInVirtualInterruptedThread(MongoInterruptedException.class,
+                    interruptDelay, "getMore", cursor::next);
+        }
+    }
+
+    @ParameterizedTest
+    @MethodSource("arguments")
+    void serverSelection(final Duration interruptDelay) {
+        // we assume here that the time to elect a new leader is greater than `interruptDelay`
+        client.getDatabase("admin").runCommand(BsonDocument.parse(
+                "{\n"
+                // `replSetStepDown` makes only the current primary an unelectabile one.
+                // We would like to make all nodes unelectable for `blockTimeout`, but there seem to be no way.
+                // It is still better to affect at least the current primary,
+                // because re-electing a different node must at least be no faster than re-electing the current one.
+                + "    'replSetStepDown': " + blockTimeout.toSeconds() + ",\n"
+                + "    'force': true\n"
+                + "}"));
+        MongoIterable<String> iterable = client.listDatabaseNames();
+        assertThrowsInVirtualInterruptedThread(MongoInterruptedException.class,
+                interruptDelay, null, iterable::first);
+    }
+
+    @ParameterizedTest
+    @MethodSource("arguments")
+    void connectionCheckout(final Duration interruptDelay) throws InterruptedException {
+        Future<?> backgroundConnectionBlocker = ForkJoinPool.commonPool().submit(() ->
+                withBlockedCommand("listIndexes", blockTimeout, () -> collection.listIndexes().first()));
+        // Wait until the background task blocks the only connection in the pool.
+        // We need the duration the connection remains blocked after this waiting to be greater than `interruptDelay`.
+        Thread.sleep(blockTimeout.minus(interruptDelay).dividedBy(2));
+        assertThrowsInVirtualInterruptedThread(MongoInterruptedException.class,
+                interruptDelay, null, collection::drop);
```

Comment:
If you either replace `() -> collection.listIndexes().first()` with `() -> null` (i.e., stop blocking the single connection there is), or replace `maxSize(1)` with `maxSize(2)` in the `beforeAll` method, this test fails with `org.opentest4j.AssertionFailedError: Expected com.mongodb.MongoInterruptedException to be thrown, but nothing was thrown.`, as expected.